packages/components/src/components/tooltip/tooltip.js

```diff
@@ -151,71 +189,176 @@ class Tooltip extends mixin(
+              );
+            }
+          }
+
+          const hasFocusin =
+            'onfocusin' in this.element.ownerDocument.defaultView;
+          const focusinEventName = hasFocusin ? 'focusin' : 'focus';
+          this._handleFocusinListener = this.manage(
+            on(
+              this.element.ownerDocument,
+              focusinEventName,
+              this._handleFocusin,
+              !hasFocusin
+            )
+          );
+        } else {
+          this.element.focus();
+        }
+
+        if (typeof callback === 'function') callback();
+      }
     );
   }
 
   /**
-   * Attaches event handlers to show/hide the tooltip.
+   * Attaches event handlers to show the tooltip.
    * @param {Element} element The element to attach the events to.
    * @private
    */
   _hookOn(element) {
-    const hasFocusin = 'onfocusin' in window;
-    const focusinEventName = hasFocusin ? 'focusin' : 'focus';
-    [focusinEventName, 'blur', 'touchleave', 'touchcancel'].forEach(name => {
+    /**
+     * Setup the _debouncedHandleClick function for displaying a tooltip
+     * @param {Event} evt - user initiated event
+     * @param {Integer[]} [allowedKeys] - allowed key codes the user may press to open the tooltip
+     * @private
+     */
+    const setupDebounceClick = (evt, allowedKeys) => {
+      const { relatedTarget, type, which } = evt;
+      // Allow user to use `space` or `enter` to open tooltip
+      if (typeof allowedKeys === 'undefined' || allowedKeys.includes(which)) {
+        const hadContextMenu = this._hasContextMenu;
+        this._hasContextMenu = type === 'contextmenu';
+        this._debouncedHandleClick({
+          relatedTarget,
+          type,
+          hadContextMenu,
+          details: getLaunchingDetails(evt),
+        });
+      }
+    };
+
+    this.manage(on(element, 'click', setupDebounceClick, false));
+
+    if (this.element.tagName !== 'BUTTON') {
       this.manage(
         on(
-          element,
-          name,
+          this.element,
+          // Does Carbon prefer keydown or keyup?
+          'keydown',
           event => {
-            const { relatedTarget, type } = event;
-            const hadContextMenu = this._hasContextMenu;
-            this._hasContextMenu = type === 'contextmenu';
-            this._debouncedHandleClick({
-              relatedTarget,
-              type: type === 'focusin' ? 'focus' : type,
-              hadContextMenu,
-              details: getLaunchingDetails(event),
-            });
+            setupDebounceClick(event, this._allowedOpenKeys);
           },
-          name === focusinEventName && !hasFocusin
+          false
         )
       );
-    });
+    }
+  }
+
+  /**
+   * Attaches event handlers to hide the tooltip.
+   * @param {Element} element The element to attach the events to.
+   * @private
+   */
+  _hookCloseActions(element) {
+    this._handleClickListener = this.manage(
+      on(element.ownerDocument, 'click', evt => {
```

Comment:
In this scenario (close on click outside) the tooltip body loses focus and thus tooltip automatically closes.

Reply:
@elizabethsjudd Any words?

Reply:
@asudoh So i took this out and it is NOT working as you're describing.  I've tried various other setups (including backing back the original focus/blur and limiting when it's actually triggered the open) to try and utilize the blur setup in floating menu, all without any luck.  Without more specifics about what I would have to update in my code to be able to remove this, I don't know what else to do other than to leave this event listener.  It is only applied while the tooltip is opened so it has very minimal impact on performance. Per your other comments, I've also tried moving the setup for this event listener to multiple different locations instead of having it in the current anonymous function block again without any success because of how tooltip is currently overwriting floating menu.

Reply:
My apologies if my guidance was not clear enough. If you have any questions or anything unclear wrt my comments or the codebase, please don't hesitate to just ask!

Here's what I found in your latest code. One reason why focus-management approach didn't work was that you seem to have taken out [the corresponding handler in `tooltip.js`](https://github.com/carbon-design-system/carbon/blob/v10.3.0/packages/components/src/components/tooltip/tooltip.js#L197-L217). The focus-management code [is still in `floating-menu.js`](https://github.com/carbon-design-system/carbon/blob/v10.3.0/packages/components/src/components/floating-menu/floating-menu.js#L141-L156), however, I found a bug where the code doesn't work when "focus is lost bug no element got focus" scenario. Adding the following code to [`track-blur.js`](https://github.com/carbon-design-system/carbon/blob/v10.3.0/packages/components/src/globals/js/mixins/track-blur.js) makes it back working. Would you try?

```diff
--- a/packages/components/src/globals/js/mixins/track-blur.js
+++ b/packages/components/src/globals/js/mixins/track-blur.js
@@ -20,6 +20,7 @@ function trackBlur(ToMix) {
       super(element, options);
       const hasFocusin = 'onfocusin' in window;
       const focusinEventName = hasFocusin ? 'focusin' : 'focus';
+      const focusoutEventName = hasFocusin ? 'focusout' : 'blur';
       this.manage(
         on(
           this.element.ownerDocument,
@@ -32,6 +33,18 @@ function trackBlur(ToMix) {
           !hasFocusin
         )
       );
+      this.manage(
+        on(
+          this.element.ownerDocument,
+          focusoutEventName,
+          event => {
+            if (!event.relatedTarget) {
+              this.handleBlur(event);
+            }
+          },
+          !hasFocusin
+        )
+      );
     }
 
     /**
```